packages/block-library/src/navigation-link/edit.js

```diff
@@ -565,10 +573,24 @@ export default function NavigationLinkEdit( {
 								// If there is no link then remove the auto-inserted block.
 								// This avoids empty blocks which can provided a poor UX.
 								if ( ! url ) {
-									// Need to handle refocusing the Nav block or the inserter?
+									// Select the previous block to keep focus nearby
+									selectPreviousBlock( clientId, true );
```

Comment:
Perhaps this line is causing the main inserter to not open when you click the "Browse All" button (See #61361).

My guess is that this occurs for the following reasons:

- The main inserter opens when "Browse All" is clicked ([See](https://github.com/WordPress/gutenberg/blob/76851ca1dd07eabe5aba30c1912e29190d26e60f/packages/block-editor/src/components/inserter/quick-inserter.js#L84-L88)).
- `onClose` is called because the focus leaves the popover.
- Since the URL does not exist, this part will be executed and the previous block will be selected.
- The main inserter automatically closes because the block is selected.

I don't understand everything about this PR, but I would suggest [not binding `onSelectBlock` directly to the popover component's `onClose` prop](https://github.com/WordPress/gutenberg/blob/76851ca1dd07eabe5aba30c1912e29190d26e60f/packages/block-library/src/navigation-link/link-ui.js#L211-L213). It might be better to clearly separate `onClose` and `onSelectBlock`, and only move the focus to the previous block when `onSelectBlock` is called.

I haven't tested it thoroughly, but the code below works fine as far as I've tested it. I hope this helps.

```diff
diff --git a/packages/block-library/src/navigation-link/edit.js b/packages/block-library/src/navigation-link/edit.js
index 4902a6cb2b..cb89881f79 100644
--- a/packages/block-library/src/navigation-link/edit.js
+++ b/packages/block-library/src/navigation-link/edit.js
@@ -562,6 +562,15 @@ export default function NavigationLinkEdit( {
                                                        clientId={ clientId }
                                                        link={ attributes }
                                                        onClose={ () => {
+                                                               // When onClose is called, it means that the "Browse all" button
+                                                               // was clicked and the main inserter was opened. In this case,
+                                                               // remove the empty block and close the link UI. It does not return
+                                                               // focus to the previous block to prevent the main inserter from
+                                                               // closing.
+                                                               onReplace( [] );
+                                                               setIsLinkOpen( false );
+                                                       } }
+                                                       onSelectBlock={ () => {
                                                                // If there is no link then remove the auto-inserted block.
                                                                // This avoids empty blocks which can provided a poor UX.
                                                                if ( ! url ) {
diff --git a/packages/block-library/src/navigation-link/link-ui.js b/packages/block-library/src/navigation-link/link-ui.js
index ce79af40e4..64a742e776 100644
--- a/packages/block-library/src/navigation-link/link-ui.js
+++ b/packages/block-library/src/navigation-link/link-ui.js
@@ -208,10 +208,6 @@ export function LinkUI( props ) {
                `link-ui-link-control__description`
        );
 
-       // Selecting a block should close the popover and also remove the (previously) automatically inserted
-       // link block so that the newly selected block can be inserted in its place.
-       const { onClose: onSelectBlock } = props;
-
        return (
                <Popover
                        placement="bottom"
@@ -291,7 +287,7 @@ export function LinkUI( props ) {
                                                setAddingBlock( false );
                                                setFocusAddBlockButton( true );
                                        } }
-                                       onSelectBlock={ onSelectBlock }
+                                       onSelectBlock={ props.onSelectBlock }
                                />
                        ) }
                </Popover>
```

Reply:
Yes, when I made this change locally and ran the E2E tests, some tests failed 😅 There may be a better approach.